File: 2816.cpp
struct Solution {
    ListNode* doubleIt(ListNode* head) {
        if (head->val >= 5){
            head = new ListNode(0, head);
        }
        head->val <<= 1;
        ListNode *prev = head, *ptr = head->next;
        while (ptr != nullptr){
            ptr->val <<= 1;
            if (ptr->val >= 10){
                ++(prev->val);
                ptr->val -= 10;
            }
            prev = ptr;
            ptr = ptr->next;
        }
        return head;
    }
};

File: sol/2816/2816.py
class Solution:
    def doubleIt(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head.val >= 5:
            head = ListNode(0, head)
        head.val *= 2
        prev = head
        ptr = head.next
        while ptr is not None:
            ptr.val *= 2
            if ptr.val >= 10:
                prev.val += 1
                ptr.val -= 10
            prev, ptr = ptr, ptr.next
        return head
